frontend/src/app/components/block/block.component.ts

```diff
@@ -295,6 +295,7 @@ export class BlockComponent implements OnInit, OnDestroy {
             ),
           !this.isAuditAvailableFromBlockHeight(block.height) ? of(null) : this.apiService.getBlockAudit$(block.id)
             .pipe(
+              tap(() => this.cacheService.setBlockAuditLoaded(block.id)),
```

Comment:
let's move this cache tracking stuff inside the `api.getBlockAudit$()` method so we don't need to remember to add this `tap()` section every time we use the API endpoint.

Reply:
I did this initially, but ran into circular dependancy issue (`apiService` is imported in `cacheService`)
Not sure how to get around it 😅

Reply:
hmm, I guess either move the entry point into `cacheService`, and fetch audits like `cacheService.getBlockAudit$()`, or move the *cache* part into `apiService`.